cgen: use of methods requiring temporary variable before and after an anonymous function

**V version:** V 0.1.27 85c7415
**OS:** 5.4.43-1-MANJARO

**What did you do?**
Use a function that requires a temporary variable before AND after the use of an anonymous function.
```v
fn main() {
    arr := [1, 2, 3]
    arr.map(it + 1) // any instruction that requires a temporary variable
    f := fn (i int) int { return i + 1 } 
    arr.map(it + 1) // any instruction that requires a temporary variable
}
```

**What did you expect to see?**
Successful compilation.
 
**What did you see instead?**
```
      |  ^~~
/home/gladear/.cache/v/test.tmp.c:9418:6: error: redefinition of ‘_t1_len’
 9418 |  int _t1_len = arr.len;
      |      ^~~~~~~
/home/gladear/.cache/v/test.tmp.c:9408:6: note: previous definition of ‘_t1_len’ was here
 9408 |  int _t1_len = arr.len;
      |      ^~~~~~~
/home/gladear/.cache/v/test.tmp.c:9419:12: error: redefinition of ‘_t1’
 9419 |  array_int _t1 = __new_array(0, _t1_len, sizeof(int));
      |            ^~~
/home/gladear/.cache/v/test.tmp.c:9409:12: note: previous definition of ‘_t1’ was here
 9409 |  array_int _t1 = __new_array(0, _t1_len, sizeof(int));
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

If you were not working with C interop, please raise an issue on GitHub:

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang
```
The reason is probably that the temporary variable counter is reset after the anonymous function declaration.

--- a/vlib/builtin/array_test.v
+++ b/vlib/builtin/array_test.v
@@ -596,6 +596,18 @@ fn test_anon_fn_map() {
 	assert [1,2,3].map(add_num) == [2,3,4]
 }
 
+fn test_mutli_anon_fn_map() {
+	a := [1,2,3].map(fn (i int) int {
+		return i + 1
+	})
+
+	b := [1,2,3].map(fn (i int) int {
+		return i + 2
+	})
+	assert a == [2,3,4]
+	assert b == [3,4,5]
+}
+
 fn test_anon_fn_arg_map() {
 	a := [1,2,3].map(fn (i int) int {
 		return i + 1

--- a/vlib/v/gen/fn.v
+++ b/vlib/v/gen/fn.v
@@ -49,7 +49,6 @@ fn (mut g Gen) gen_fn_decl(it ast.FnDecl) {
 		eprintln('INFO: compile with `v -live $g.pref.path `, if you want to use the [live] function ${it.name} .')
 	}
 	//
-	g.reset_tmp_count()
 	if is_main {
 		if g.pref.os == .windows {
 			if g.is_gui_app() {

